test/benchmarks/ml/vllm_test.go

```diff
@@ -0,0 +1,174 @@
+			defer clientCtr.CleanUp(ctx)
+
+			b.StartTimer()
+
+			// store vllm logs here
+			logsDir := b.TempDir()
+
+			out, err := clientCtr.Run(ctx, dockerutil.RunOpts{
+				Links: []string{serverCtr.MakeLink("vllmctr")},
+				Image: "benchmarks/vllm",
+				Env:   []string{"PYTHONPATH=$PYTHONPATH:/vllm"},
+				Mounts: []mount.Mount{
+					// The logs dir is used because vllm only outputs json to a file.
+					{
+						Source: logsDir,
+						Target: "/tmp",
+						Type:   "bind",
+					},
+				},
+			}, "/vllm/benchmarks/benchmark_serving.py", "--host", "vllmctr", "--model", "/model", "--tokenizer", "/model", "--endpoint", "/v1/completions", "--backend", "openai", "--dataset", "/ShareGPT_V3_unfiltered_cleaned_split.json", "--save-result", "--result-dir", "/tmp")
+			if err != nil {
+				b.Errorf("failed to run container: %v logs: %s", err, out)
+			}
+
+			b.StopTimer()
+
+			metrics, err := parseVLLMJSON(logsDir)
+			if err != nil {
+				b.Errorf("failed to parse vllm output: %v", err)
+			}
+			b.ReportMetric(float64(metrics.Completed), "requests")
+			b.ReportMetric(metrics.RequestThroughput, "request_throughput")
+			b.ReportMetric(metrics.InputThroughput, "input_tok_throughput")
+			b.ReportMetric(metrics.OutputThroughput, "output_tok_throughput")
+			b.ReportMetric(metrics.MedianTTFTMS, "median_ttft_ms")
+			b.ReportMetric(metrics.MediaTPOTMS, "median_tpot_ms")
+		}
+	})
+}
+
+// Modeled after the metrics reported here: https://github.com/vllm-project/vllm/blob/main/benchmarks/benchmark_serving.py#L338-L358
+type metrics struct {
+	Duration          float64 `json:"duration"`
+	Completed         int     `json:"completed"`
+	TotalInputTokens  int     `json:"total_input_tokens"`
+	TotalOutputTokens int     `json:"total_output_tokens"`
+	RequestThroughput float64 `json:"request_throughput"`
+	InputThroughput   float64 `json:"input_throughput"`
+	OutputThroughput  float64 `json:"output_throughput"`
+	MeanTTFTMS        float64 `json:"mean_ttft_ms"`
+	MedianTTFTMS      float64 `json:"median_ttft_ms"`
+	P99TTFTMS         float64 `json:"p99_ttft_ms"`
+	MeanTPOTMS        float64 `json:"mean_tpot_ms"`
+	MediaTPOTMS       float64 `json:"median_tpot_ms"`
+	P99TPOTMS         float64 `json:"p99_tpot_ms"`
+	// metrics that are available but unused so far.
+	// InputLens         []int   `json:"input_lens"`
+	// OutputLens        []int   `json:"output_lens"`
+	// TTFTs             []float64   `json: "ttfts"`
+	// ITLS              [][]float64 `json: "itls"`
+	// GeneratedTexts    float64 `json: "generated_texts"`
+	// Errors            float64 `json: "errors"`
+}
+
+// parseVLLMJSON expects a path that contains only one json file.
+func parseVLLMJSON(path string) (metrics, error) {
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return metrics{}, fmt.Errorf("failed to read directory: %w", err)
+	}
+
+	var jsonPath string
+	for _, name := range files {
+		if strings.HasSuffix(name.Name(), ".json") {
```

Comment:
What if there are multiple such files? Should it read all of them? Should it error out? Please handle this case explicitly.